conbench/tests/api/test_runs.py

```diff
@@ -231,6 +231,27 @@ def test_closest_commit_different_machines_should_not_prefer_test_runs_as_baseli
         response = client.get(f"/api/runs/{contender_run.id}/")
         self.assert_200_ok(response, _expected_entity(contender_run, baseline_run.id))
 
+    def test_baseline_run_always_on_default_branch(self, client):
+        self.authenticate(client)
+        _, benchmark_results = _fixtures.gen_fake_data()
+
+        successes = 0
+        for benchmark_result in benchmark_results:
+            response = client.get(f"/api/runs/{benchmark_result.run_id}/")
+            assert response.status_code == 200, response.status_code
+            baseline_run_link = response.json["links"].get("baseline")
+            if baseline_run_link:
+                baseline_run_response = client.get(
+                    baseline_run_link.replace("http://localhost", "")
+                )
+                assert (
+                    baseline_run_response.status_code == 200
+                ), baseline_run_response.status_code
+                assert baseline_run_response.json["commit"]["branch"] == "default"
+                successes += 1
+
+        assert successes == 10
+
```

Comment:
Do we already have tests (possibly up above, I skimmed the names but didn't see any that clearly matched this):

* Contender is on a branch and the fork-point has benchmarks (so can be used)
* Contender is on a branch and the fork-point has no benchmarks (so we use the parent of the fork-point if it has benchmarks)
* Contender is on a branch, the parent of the contender has benchmarks but we ignore those and use the fork-point
* Contender is on default, and we use the parent (if it has benchmarks)
* Contender is on default, and the parent doesn't have benchmarks (so we use the grandparent if that has benchmarks)

Reply:
Yes, these cases are all covered in [gen_fake_data()](https://github.com/conbench/conbench/blob/61fac4e5c12ee4b71a493929662e580084e4186b/conbench/tests/api/_fixtures.py#L265). This means they're being tested in both [TestRunGet.test_baseline_run_always_on_default_branch](https://github.com/conbench/conbench/blob/e57612363845b55d6d8072661c9101277ae53138/conbench/tests/api/test_runs.py#L234) for "is their baseline on the default branch?" and [test_get_baseline_run](https://github.com/conbench/conbench/blob/e57612363845b55d6d8072661c9101277ae53138/conbench/tests/entities/test_run.py#L19) for the explicit behavior you called out.

- Contender is on a branch and the fork-point has benchmarks (so can be used) -> fake benchmark results 2, 3, 6, 7
- Contender is on a branch and the fork-point has no benchmarks (so we use the parent of the fork-point if it has benchmarks) -> fake benchmark result 4
- Contender is on a branch, the parent of the contender has benchmarks but we ignore those and use the fork-point -> fake benchmark result 7
- Contender is on default, and we use the parent (if it has benchmarks) -> fake benchmark result 1
- Contender is on default, and the parent doesn't have benchmarks (so we use the grandparent if that has benchmarks) -> fake benchmark results 5, 8, 9, 16

(Also, fake benchmark result 0 has no parents, and fake benchmark results 10-15 can't be compared for one reason or another.)

Reply:
Super good discussion.